client/src/utils/propositions.ts

```diff
@@ -0,0 +1,169 @@
+ *
+ * @param prop - Proposition object (may be undefined)
+ * @returns True if the proposition has a geneContextQualifier property
+ */
+export function hasGeneContextQualifier(
+  prop:
+    | ExperimentalVariantFunctionalImpactProposition
+    | VariantDiagnosticProposition
+    | VariantOncogenicityProposition
+    | VariantPathogenicityProposition
+    | VariantPrognosticProposition
+    | VariantTherapeuticResponseProposition
+    | undefined,
+): prop is Exclude<typeof prop, ExperimentalVariantFunctionalImpactProposition | undefined> & {
+  geneContextQualifier?: { extensions?: { name: string; value: unknown }[] }
+} {
+  return !!prop && 'geneContextQualifier' in prop
+}
+
+/**
+ * Extracts therapy information from a proposition.
+ * Only VariantTherapeuticResponseProposition objects have therapies.
+ *
+ * @param prop - Proposition object (may be undefined)
+ * @returns Therapy name string, or "N/A" if not applicable
+ */
+export function getTherapyFromProposition(
+  prop:
+    | VariantTherapeuticResponseProposition
+    | VariantDiagnosticProposition
+    | VariantPrognosticProposition
+    | VariantOncogenicityProposition
+    | VariantPathogenicityProposition
+    | ExperimentalVariantFunctionalImpactProposition
+    | undefined,
+): string {
+  if (!prop) return 'N/A'
+
+  if ('objectTherapeutic' in prop) {
+    const therapeutic = prop.objectTherapeutic
+    return typeof therapeutic !== 'string' ? (formatTherapies(therapeutic) ?? 'N/A') : 'N/A'
+  }
+
+  return 'N/A'
+}
+
+/**
+ * Type guard to determine if a Therapeutic is a TherapyGroup.
+ */
+function isTherapyGroup(obj: Therapeutic): obj is TherapyGroup {
+  return Array.isArray((obj as TherapyGroup).therapies)
+}
+
+/**
+ * Formats a Therapeutic into a human-readable therapy string.
+ * - For TherapyGroup, joins multiple therapies with "and"/"or"
+ * - For MappableConcept of type "Therapy", returns its name
+ *
+ * @param objectTherapeutic - Therapeutic object to format
+ * @returns Formatted therapy name or null if not available
+ */
+const formatTherapies = (objectTherapeutic: Therapeutic): string | null => {
+  if (!objectTherapeutic) return null
+
+  if (isTherapyGroup(objectTherapeutic)) {
+    // It's a TherapyGroup
+    const names = objectTherapeutic.therapies.map((t) => t?.name).filter(Boolean)
+    if (names.length === 0) return null
+    if (names.length === 1) return names[0] ?? null
+
+    const operator = objectTherapeutic.membershipOperator?.toLowerCase() === 'or' ? 'or' : 'and'
+    return `${names.slice(0, -1).join(', ')} ${operator} ${names[names.length - 1]}`
+  }
+
+  // Otherwise it's a MappableConcept
+  if (objectTherapeutic.conceptType === 'Therapy') {
+    return objectTherapeutic.name ?? null
+  }
+
+  return null
+}
+
+/**
+ * Extracts human-readable condition names from a Condition object.
+ *
+ * @param condition - A string, MappableConcept, ConditionSet, or undefined
+ * @returns Array of condition names (may be multiple for ConditionSet)
```

Comment:
@jsstevenson this is what we're currently discussing in the meeting - here's how I decided to handle condition sets - I make every condition a set, whether it's a single condition or a conditionset. Then, these get flattened to create the filters later

Reply:
Works for now. I know that coming down the pike we have cases like "Cancer AND (baby onset OR teenage onset)" but we don't necessarily need to have perfect solutions for those cases immediately